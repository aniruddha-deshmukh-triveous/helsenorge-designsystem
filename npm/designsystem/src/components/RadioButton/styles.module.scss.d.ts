export type Styles = {
  'radio-button': string;
  'radio-button--disabled': string;
  'radio-button--invalid': string;
  'radio-button--on-blueberry': string;
  'radio-button--on-dark': string;
  'radio-button-errors': string;
  'radio-button-label': string;
  'radio-button-label--bigform': string;
  'radio-button-label--disabled': string;
  'radio-button-label--invalid': string;
  'radio-button-label--on-blueberry': string;
  'radio-button-label--on-dark': string;
  'radio-button-wrapper': string;
  'radio-button-wrapper--bigform': string;
  'radio-button-wrapper--with-error': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
