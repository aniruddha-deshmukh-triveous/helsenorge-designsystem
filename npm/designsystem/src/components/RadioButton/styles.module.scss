@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_palette.scss';
@import '../../scss/font-settings.scss';

.radio-button-errors {
  font-size: $font-size-sm;
  font-weight: 600;
  color: $cherry500;
  margin-top: 0;
}

.radio-button-wrapper {
  cursor: pointer;
  margin: getSpacer(s) 0;

  &--with-error {
    padding: getSpacer(s) getSpacer(xs) getSpacer(m);
    background-color: $cherry50;
    margin-left: -0.75rem;
    border-left: 0.25rem solid $cherry500;

    @media (min-width: map-get($grid-breakpoints, md)) {
      padding: getSpacer(s) getSpacer(m) getSpacer(l);
    }
  }

  &--with-error#{&}--bigform {
    padding: getSpacer(m) getSpacer(m);
  }

  &--bigform {
    margin: getSpacer(xs) 0 getSpacer(xs) 0;
  }
}

.radio-button-wrapper:first-of-type {
  margin-top: 0;
}

.radio-button-wrapper:last-of-type {
  margin-bottom: 0;
}

.radio-button-label {
  display: flex;
  align-items: center;
  cursor: pointer;
  font-size: $font-size-sm;

  @media (min-width: map-get($grid-breakpoints, md)) {
    font-size: $font-size-md;
  }

  &--on-dark {
    color: $white;
  }

  &--invalid {
    color: $black;
  }

  &--disabled {
    cursor: default;
    color: $neutral400;
  }

  &--bigform {
    padding: getSpacer(s);
    background-color: $neutral50;
  }

  &--bigform#{&}--on-blueberry {
    background-color: $blueberry50;
  }

  &--bigform#{&}--on-dark {
    background-color: $blueberry800;
  }

  &--bigform#{&}--invalid {
    background-color: $white;
  }

  @media (min-width: map-get($grid-breakpoints, md)) {
    &--bigform {
      padding: 1.68rem getSpacer(l);
    }
  }
}

.radio-button {
  -moz-appearance: none;
  -webkit-appearance: none;
  -o-appearance: none;

  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  border-radius: 10rem;
  margin: getSpacer(3xs) getSpacer(s) getSpacer(3xs) getSpacer(3xs);
  height: 1.5rem;
  width: 1.5rem;
  min-width: auto !important;
  min-height: auto !important;
  color: $neutral700;
  box-shadow: 0 0 0 getSpacer(3xs);

  &:checked {
    color: $blueberry500;

    &:hover {
      background-color: $blueberry50;
    }
  }

  &:focus {
    box-shadow: 0 0 0 getSpacer(2xs);
    color: $black;
  }

  &:hover {
    box-shadow: 0 0 0 getSpacer(2xs);
    background-color: $neutral100;
  }

  :active > & {
    box-shadow: 0 0 0 getSpacer(2xs);
    color: $black;
  }

  &--on-blueberry {
    color: $blueberry500;

    :hover > & {
      background-color: $blueberry100;
    }
  }

  &--on-dark {
    color: $neutral200;

    &:hover {
      background-color: $inverted-hover;
      box-shadow: 0 0 0 getSpacer(2xs);
    }

    &:checked {
      color: $blueberry200;

      &:hover {
        background-color: $inverted-hover;
      }
    }

    :active > & {
      box-shadow: 0 0 0 getSpacer(2xs);
      color: $white;
    }

    &:focus {
      color: $white;
    }
  }

  &--invalid {
    color: $cherry500;

    &:hover {
      background-color: $cherry100;
    }

    &:checked {
      color: $cherry500;

      &:hover {
        background-color: $cherry100;
      }
    }

    &:active {
      color: $black;
    }

    &:focus {
      color: $black;
    }
  }

  &--disabled {
    cursor: default;

    color: $neutral400;
    &:checked {
      color: $neutral400;

      &:hover {
        box-shadow: 0 0 0 getSpacer(3xs);
        background-color: transparent;
      }
    }
    &:hover {
      box-shadow: 0 0 0 getSpacer(3xs);
      background-color: transparent;
    }

    &:active {
      color: $neutral400;
    }
  }

  // :before er dot i RadioButton
  &:checked::before {
    content: '';
    display: flex;
    background-color: $blueberry500;
    width: 0.75rem;
    height: 0.75rem;
    border-radius: 10rem;
  }

  &--on-dark:checked::before {
    background-color: $blueberry200;
  }

  &--disabled:checked::before {
    box-shadow: 0 0 0 0;
    background-color: $neutral400;
    color: $neutral400;
  }

  &--invalid:checked::before {
    background-color: $cherry500;
  }
}
