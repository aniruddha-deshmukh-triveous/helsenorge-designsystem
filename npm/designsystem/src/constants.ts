export enum IconSize {
  XSmall = 38,
  Small = 48,
  Medium = 64,
  Large = 80,
  XLarge = 130,
}

export enum ZIndex {
  Modal = 1300000,
}

export interface HTMLButtonProps {
  autoFocus?: boolean;
  disabled?: boolean;
  form?: string;
  formAction?: string;
  formEncType?: string;
  formMethod?: string;
  formNoValidate?: boolean;
  formTarget?: string;
  name?: string;
  type?: 'submit' | 'reset' | 'button' | string;
  value?: string | string[] | number;
}

export interface HTMLTextareaProps {
  autoFocus?: boolean;
  disabled?: boolean;
  form?: string;
  formNoValidate?: boolean;
  formTarget?: string;
  name?: string;
  id?: string;
  autocomplete?: 'off' | 'on';
  placeholder?: string;
  readonly?: boolean;
  required?: boolean;
  spellcheck?: 'true' | 'default' | 'false';
  'aria-invalid'?: boolean;
}

export interface HTMLAnchorProps {
  download?: string;
  href?: string;
  hrefLang?: string;
  media?: string;
  ping?: string;
  rel?: string;
  target?: string;
  referrerPolicy?:
    | ''
    | 'same-origin'
    | 'no-referrer'
    | 'no-referrer-when-downgrade'
    | 'origin'
    | 'origin-when-cross-origin'
    | 'strict-origin'
    | 'strict-origin-when-cross-origin'
    | 'unsafe-url';
}

export type ButtonVariant = 'secondary' | 'tertiary' | string | undefined | null;

export enum FormMode {
  onwhite = 'onwhite',
  ongrey = 'ongrey',
  onblueberry = 'onblueberry',
  ondark = 'ondark',
  oninvalid = 'oninvalid',
}

export enum FormVariant {
  normal = 'normal',
  bigform = 'bigform',
}
