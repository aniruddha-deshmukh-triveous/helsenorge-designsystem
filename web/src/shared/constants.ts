export const HNDS_BUCKET_IMAGE_URL = ''; // Kan endres her hvis bilder skal hostes et annet sted

export interface StyledDefaultProps {
  children?: React.ReactNode;
  className?: string;
}
