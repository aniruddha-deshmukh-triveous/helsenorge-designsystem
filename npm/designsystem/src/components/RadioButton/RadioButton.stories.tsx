import React from 'react';
import { boolean, select, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import RadioButton from './RadioButton';
import { FormMode, FormVariant } from '../../constants';

const stories = storiesOf('RadioButton', module);
stories.addDecorator(withKnobs);

stories.add('Default', () => (
  <div
    style={{
      width: '40rem',
    }}
  >
    <RadioButton
      label={'Check me out!'}
      defaultChecked={boolean('Checked', false)}
      disabled={boolean('Disabled', false)}
      mode={select('Mode', FormMode, FormMode.OnWhite)}
      variant={select('Variant', FormVariant, FormVariant.Normal)}
    />
  </div>
));
