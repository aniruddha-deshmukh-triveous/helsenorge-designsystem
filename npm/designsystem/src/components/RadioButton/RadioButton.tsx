import React from 'react';

import classNames from 'classnames';

import { AnalyticsId, FormMode, FormVariant } from '../../constants';
import { uuid } from '../../utils/uuid';

import radioButtonStyles from './styles.module.scss';

export interface RadioButtonProps
  extends Pick<React.InputHTMLAttributes<HTMLInputElement>, 'name' | 'value' | 'disabled' | 'defaultChecked' | 'required'> {
  /** Adds custom classes to the element. */
  className?: string;
  /** The label text next to the radioButton */
  label: string;
  /** input id of the radioButton */
  inputId?: string;
  /** Changes the visuals of the radioButton */
  mode?: keyof typeof FormMode;
  /** Changes the visuals of the radioButton */
  variant?: keyof typeof FormVariant;
  /** Activates Error style for the radioButton - This is can be true while errorText is empty, when in a FormGroup */
  error?: boolean;
  /** Error text to show above the component */
  errorText?: string;
  /** Sets the data-testid attribute. */
  testId?: string;
}

export const RadioButton = React.forwardRef((props: RadioButtonProps, ref: React.Ref<HTMLInputElement>) => {
  const {
    className,
    defaultChecked = false,
    disabled,
    label,
    inputId = uuid(),
    mode,
    name = inputId,
    variant,
    errorText,
    error = !!errorText,
    value = label,
    testId,
    required,
    ...rest
  } = props;
  const invalid = error || mode === FormMode.oninvalid;
  const onDark = mode === FormMode.ondark;
  const onBlueberry = mode === FormMode.onblueberry;
  const bigform = variant === FormVariant.bigform;

  const radioButtonWrapperClasses = classNames(radioButtonStyles['radio-button-wrapper'], {
    [radioButtonStyles['radio-button-wrapper--with-error']]: errorText,
    [radioButtonStyles['radio-button-wrapper--bigform']]: bigform,
  });
  const radioButtonLabelClasses = classNames(radioButtonStyles['radio-button-label'], {
    [radioButtonStyles['radio-button-label--disabled']]: disabled,
    [radioButtonStyles['radio-button-label--on-dark']]: onDark,
    [radioButtonStyles['radio-button-label--on-blueberry']]: onBlueberry,
    [radioButtonStyles['radio-button-label--invalid']]: invalid,
    [radioButtonStyles['radio-button-label--bigform']]: bigform,
  });
  const radioButtonClasses = classNames(
    radioButtonStyles['radio-button'],
    {
      [radioButtonStyles['radio-button--on-dark']]: onDark,
      [radioButtonStyles['radio-button--disabled']]: disabled,
      [radioButtonStyles['radio-button--on-blueberry']]: onBlueberry,
      [radioButtonStyles['radio-button--invalid']]: invalid,
    },
    className
  );
  const errorStyles = classNames(radioButtonStyles['radio-button-errors'], {
    [radioButtonStyles['radio-button-errors--bigform']]: bigform,
  });

  return (
    <div data-testid={testId} data-analyticsid={AnalyticsId.RadioButton} className={radioButtonWrapperClasses}>
      {errorText && <p className={errorStyles}>{errorText}</p>}
      <label htmlFor={inputId} className={radioButtonLabelClasses}>
        <input
          id={inputId}
          name={name}
          className={radioButtonClasses}
          type="radio"
          disabled={disabled}
          value={value}
          ref={ref}
          aria-invalid={error}
          defaultChecked={defaultChecked}
          required={required}
          {...rest}
        />
        {label}
      </label>
    </div>
  );
});

export default RadioButton;
