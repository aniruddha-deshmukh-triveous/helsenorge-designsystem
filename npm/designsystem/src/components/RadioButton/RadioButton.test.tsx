import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import RadioButton from './RadioButton';
import { FormMode, FormVariant } from '../../constants';

describe('Gitt at RadioButton skal vises', (): void => {
  describe('Når RadioButton rendres', (): void => {
    test('Så vises RadioButton', (): void => {
      const { container } = render(<RadioButton inputid={'test01'} label={'Radio1'} />);

      expect(container).toMatchSnapshot();

      const label = screen.getByText('Radio1');
      expect(label).toBeVisible();
      expect(label).toHaveClass('radio-button-label');

      const input = screen.getByRole('radio');
      expect(input).toBeVisible();
      expect(input).toHaveClass('radio-button');
    });
  });

  describe('Når disabled er true', (): void => {
    test('Så vises RadioButton som disabled', (): void => {
      render(<RadioButton label={'Radio1'} disabled />);

      const input = screen.getByRole('radio');
      const label = screen.getByText('Radio1');
      expect(input).toBeDisabled();
      expect(label).toHaveClass('radio-button-label radio-button-label--disabled');
    });
  });

  describe('Når mode er onBlueberry', (): void => {
    test('Så vises RadioButton med onBlueberry styling', (): void => {
      render(<RadioButton label={'Radio1'} mode={FormMode.onblueberry} />);

      const label = screen.getByText('Radio1');
      expect(label).toHaveClass('radio-button-label radio-button-label--on-blueberry');
    });
  });
  describe('Når mode er onDark', (): void => {
    test('Så vises RadioButton med onDark styling', (): void => {
      render(<RadioButton label={'Radio1'} mode={FormMode.ondark} />);

      const label = screen.getByText('Radio1');
      expect(label).toHaveClass('radio-button-label radio-button-label--on-dark');
    });
  });

  describe('Når formvariant er bigform', (): void => {
    test('Så vises RadioButton med bigform styling', (): void => {
      render(<RadioButton label={'Radio1'} variant={FormVariant.bigform} />);

      const label = screen.getByText('Radio1');
      expect(label).toHaveClass('radio-button-label radio-button-label--bigform');
    });
  });

  describe('Når startChecked er true', (): void => {
    test('Så vises RadioButton som checked', (): void => {
      render(<RadioButton label={'Radio1'} defaultChecked />);

      const input = screen.getByRole('radio');
      expect(input).toHaveProperty('checked', true);
    });
  });

  describe('Når RadioButton trykkes på', (): void => {
    test('Så vises RadioButton som checked', (): void => {
      render(<RadioButton label={'Radio1'} />);

      fireEvent.click(screen.getByText('Radio1'));

      const input = screen.getByRole('radio');
      expect(input).toHaveProperty('checked', true);
    });
  });

  describe('Når RadioButton får satt error', (): void => {
    test('Så vises RadioButton med indre error styling, uten ytre error styling', (): void => {
      render(<RadioButton label={'Radio1'} error />);

      // Indre styling
      const label = screen.getByText('Radio1');

      expect(label).toHaveClass('radio-button-label radio-button-label--invalid');

      // Ytre styling
      const wrapper = label.parentElement;
      expect(wrapper).toHaveClass('radio-button-wrapper');
    });
  });

  describe('Når RadioButton får satt errorText', (): void => {
    test('Så vises RadioButton med errormelding i tilleg til indre og ytre error styling', (): void => {
      render(<RadioButton label={'Radio1'} errorText={'error error!'} />);

      expect(screen.getByText('error error!')).toBeVisible();

      // Indre styling
      const label = screen.getByText('Radio1');

      expect(label).toHaveClass('radio-button-label radio-button-label--invalid');

      // Ytre styling
      const wrapper = label.parentElement;
      expect(wrapper).toHaveClass('radio-button-wrapper radio-button-wrapper--with-error');
    });
  });
});
