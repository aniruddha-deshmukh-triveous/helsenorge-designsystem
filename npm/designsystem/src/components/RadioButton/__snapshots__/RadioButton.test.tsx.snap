// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at RadioButton skal vises Når RadioButton rendres Så vises RadioButton 1`] = `
<div>
  <div
    class="radio-button-wrapper"
  >
    <label
      class="radio-button-label"
      for="test01"
    >
      <input
        aria-invalid="false"
        class="radio-button"
        id="test01"
        name="test01"
        type="radio"
        value="Radio1"
      />
      Radio1
    </label>
  </div>
</div>
`;
